File: acmicpc/10xxx/10845.java
import java.util.*;

class BJ10845 {
	public static void main(String[]V) {
		Scanner S = new Scanner(System.in);
		Queue<Integer> A = new LinkedList<>();
		for(int T = S.nextInt(); T > 0; T--) {
			String s = S.next();
			if(s.equals("push"))
				A.offer(S.nextInt());
			if(s.equals("pop"))
				System.out.println(A.isEmpty()?-1:A.poll());
			if(s.equals("size"))
				System.out.println(A.size());
			if(s.equals("empty"))
				System.out.println(A.isEmpty()?1:0);
			if(s.equals("front"))
				System.out.println(A.isEmpty()?-1:A.peek());
			if(s.equals("back")) {
				Object[] a = A.toArray();
				System.out.println(A.isEmpty()?-1:a[A.size()-1]);
			}
		}
	}
}

File: acmicpc/10xxx/10845.cpp
#include <iostream>
#include <algorithm>
#include <list>

using namespace std;

int T, X;
string S;
list<int> Q;

int main() {
  
  ios::sync_with_stdio(0);
  cin.tie(0);
  cout.tie(0);

  cin >> T;
  while(T--) {
    cin >> S;
    if(!S.compare("push")) {
      cin >> X;
      Q.push_back(X);
    } else if(!S.compare("pop")) {
      if(Q.empty()) cout << -1;
      else {
        cout << Q.front();
        Q.pop_front();
      }
    } else if(!S.compare("size")) {
      cout << Q.size();
    } else if(!S.compare("empty")) {
      if(Q.empty()) cout << 1;
      else cout << 0;
    } else if(!S.compare("front")) {
      if(Q.empty()) cout << -1;
      else cout << Q.front();
    } else if(!S.compare("back")) {
      if(Q.empty()) cout << -1;
      else cout << Q.back();
    }
    
    if(S.compare("push")) {
      cout << "\n";
    }
  }

  return 0;
}